src/ripple/rpc/handlers/Tx.cpp

```diff
@@ -310,13 +341,39 @@ doTxJson(RPC::JsonContext& context)
 
     // Deserialize and validate JSON arguments
 
-    if (!context.params.isMember(jss::transaction))
+    TxArgs args;
+
+    if (context.params.isMember(jss::transaction) &&
+        context.params.isMember(jss::ctid))
+        // specifying both is ambiguous
         return rpcError(rpcINVALID_PARAMS);
 
-    TxArgs args;
+    if (context.params.isMember(jss::transaction))
+    {
+        uint256 hash;
+        if (!hash.parseHex(context.params[jss::transaction].asString()))
+            return rpcError(rpcNOT_IMPL);
+        args.hash = hash;
+    }
+    else if (context.params.isMember(jss::ctid))
+    {
+        auto ctid = RPC::decodeCTID(context.params[jss::ctid].asString());
+        if (!ctid)
+            return rpcError(rpcINVALID_PARAMS);
 
-    if (!args.hash.parseHex(context.params[jss::transaction].asString()))
-        return rpcError(rpcNOT_IMPL);
+        auto const [lgr_seq, txn_idx, net_id] = *ctid;
+        if (net_id != context.app.config().NETWORK_ID)
+        {
+            std::stringstream out;
```

Comment:
I personally prefer to avoid `std::stringstream`. I'd recommend:

```
return RPC::make_error(rpcWRONG_NETWORK,
    "Wrong network. You should submit this request to a node "
    "running on NetworkID: " + std::to_string(net_id));
```